src/components/MovieForm/MovieForm.tsx

```diff
@@ -45,78 +54,98 @@ export const MovieForm: React.FC<MovieFormProps> = ({
           <input
             type="text"
             id="title"
-            name="title"
-            defaultValue={initialData?.title}
             placeholder="Title"
-            required
-            className="w-full bg-[#323232] border border-[#424242] rounded 
-                       text-white px-4 py-3 placeholder-gray-500
-                       focus:outline-none focus:border-[#F65261]"
+            {...register("title", { required: "Title is required" })}
+            className={`w-full bg-[#323232] border ${
+              errors.title ? "border-red-500" : "border-[#424242]"
+            } rounded text-white px-4 py-3 placeholder-gray-500 focus:outline-none focus:border-[#F65261]`}
           />
+          {errors.title && (
+            <p className="text-red-500 text-sm">{errors.title.message}</p>
+          )}
```

Comment:
It looks quite repetitive, you could create a helper component  that would take a field's errors object and render the message in case there is an error.